src/shims/intrinsics.rs

```diff
@@ -522,6 +522,37 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 this.write_scalar(res, dest)?;
             }
 
+            "uninit" => {
+                // Check fast path: we don't want to force an allocation in case the destination is a simple value,
```

Comment:
we can be proactive and `tcx.sess.warn("go update your dependencies")`, even if rustc doesn't do that yet

Reply:
That will be non-actionable most of the time I think.

Also this is basically https://github.com/rust-lang/miri/issues/797. We should never have warning that cannot be turned off.